opam-release.sh

```diff
@@ -1,23 +1,27 @@
 #!/bin/bash
 
 set -e
+set -u
+
+git submodule init
+git submodule update
 
 VERSION_FILE=_build/default/src/camltc_version.ml
 ## Create a tarball from the current checkout.
-NAME=$(git describe --tags --dirty)
+NAME=$(git describe --tags)
 if ! git describe --tags --exact-match 2> /dev/null; then
     echo "Cowardly refusing to create a release that has not been tagged"
     exit 1
 fi
 echo "Creating release for tag: ${NAME}"
 
-rm ${VERSION_FILE}
+rm -f ${VERSION_FILE}
 dune build
 
 # Checkout a shallow copy, including submodules.
 URL=$(git remote get-url origin)
-rm -fr ${NAME}
+rm -rf ${NAME}
 git clone ${URL} --single-branch --branch ${NAME} --depth=1 --shallow-submodules --recurse-submodules ${NAME}
 cp ${VERSION_FILE} ${NAME}/src
-tar zcf ${NAME}.tar.gz --exclude-vcs --exclude-vcs-ignores ${NAME}
-rm -fr ${NAME}
+tar cf ${NAME}.tar.gz -I 'gzip -9' --exclude-vcs --exclude-vcs-ignores ${NAME}
```

Comment:
On my machine it shaves of 7Kb, which is a reduction of 0.76%. Is there another reason for doing this other than reduce the size of the generated tarball?

Reply:
no, just try to have the smallest tarball

Reply:
If you want the smallest archive, then don't do gz but xz,
but do we care?

Reply:
we do care, but as in "all ocaml package archives should use strong compression", so that we do download all ocaml software faster